### PR TITLE
net: zperf: Configurable packet size limit

### PR DESCRIPTION
**Pull request by rettichschnidi**
Allow user to adjust (artificial) size limitation of 1024 bytes.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]
LGTM

File: subsys/net/lib/zperf/zperf_udp_uploader.c [hunk 1]
```diff
@@ -160,6 +160,8 @@ void zperf_udp_upload(const struct shell *sh,
 			      PACKET_SIZE_MAX);
 		packet_size = PACKET_SIZE_MAX;
 	} else if (packet_size < sizeof(struct zperf_udp_datagram)) {
+		BUILD_ASSERT(sizeof(struct zperf_udp_datagram) <= PACKET_SIZE_MAX,
+			     "Invalid PACKET_SIZE_MAX");
```
Comment (rlubos) [inline subsys/net/lib/zperf/zperf_udp_uploader.c:164]: I'm just not sure if it's the right place for such a build assert? Wouldn't it be better to place it in `zperf_internal.h`, where `struct zperf_udp_datagram` and `PACKET_SIZE_MAX` are defined?

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

File: subsys/net/lib/zperf/zperf_udp_uploader.c (hunk 1 above)
Comment (rettichschnidi) [reply subsys/net/lib/zperf/zperf_udp_uploader.c]: Done

Unsure however about if and which header to include when using BUILD_ASSERT.

Comment (rlubos) [reply subsys/net/lib/zperf/zperf_udp_uploader.c]: The macro itself is defined in `zephyr/toolchain/common.h`, but this header is pulled by `__assert.h` so I guess it's fine.

**Pull request merged**